### ISSUE TITLE
Creditmemo not working for simple only products with no product options

### ISSUE THREAD

**Issue by 4DallasC** [magento/magento2#25694]
### Preconditions (*)

1. Magento 2.3.3

### Steps to reproduce (*)
1. Create Order with simple only products and no options.
2. Create Creditmemo - save button give javascript errors - disableElements not defined
3. Orders with product options bring in script code that requires prototype. Bring in the legacy-build.min.js javascript library solves this issue.

### Expected result (*)

1. Refund button is enable and Creditmeno is created

### Actual result (*)

1. javascript error prevents the creation of the creditmemo

**Comment by 4DallasC** [comment]
2.3-develop has the same old JavaScript code in the onclick event to create the creditmemo.
#confirmed

**Comment by engcom-Echo** [comment]
Hello @4DallasC 

Could you provide more details for reproduce this issue?
Please add screenshots.

Thank you!

**Comment by 4DallasC** [comment]
Howdy @engcom-Echo 

In the files:-
- vendor/magento/module-sales/Block/Adminhtml/Order/Creditmemo/Create/Items.php
- vendor/magento/module-sales/Block/Adminhtml/Order/Invoice/Create/Items.php
- vendor/magento/module-shipping/view/adminhtml/templates/create/items.phtml

All the above files include the same outdated JavaScript function without using require(['legacy-build.min']); or require(['prototype']); as seen below. This inclusions are required to inject the function into the page and the error is caused when it is not.

```
namespace Magento\Sales\Block\Adminhtml\Order\Creditmemo\Create;
class Items extends \Magento\Sales\Block\Adminhtml\Items\AbstractItems
{
    protected function _prepareLayout()
    {
        $onclick = "submitAndReloadArea($('creditmemo_item_container'),'" . $this->getUpdateUrl() . "')";
        $this->addChild(
            'update_button',
            \Magento\Backend\Block\Widget\Button::class,
            ['label' => __('Update Qty\'s'), 'class' => 'update-button', 'onclick' => $onclick]
        );

        if ($this->getCreditmemo()->canRefund()) {
            if ($this->getCreditmemo()->getInvoice() && $this->getCreditmemo()->getInvoice()->getTransactionId()) {
                $this->addChild(
                    'submit_button',
                    \Magento\Backend\Block\Widget\Button::class,
                    [
                        'label' => __('Refund'),
                        'class' => 'save submit-button refund primary',
                        'onclick' => 'disableElements(\'submit-button\');submitCreditMemo()'
                    ]
                );
            }
            $this->addChild(
                'submit_offline',
                \Magento\Backend\Block\Widget\Button::class,
                [
                    'label' => __('Refund Offline'),
                    'class' => 'save submit-button primary',
                    'onclick' => 'disableElements(\'submit-button\');submitCreditMemoOffline()'
                ]
            );
        } else {
            $this->addChild(
                'submit_button',
                \Magento\Backend\Block\Widget\Button::class,
                [
                    'label' => __('Refund Offline'),
                    'class' => 'save submit-button primary',
                    'onclick' => 'disableElements(\'submit-button\');submitCreditMemoOffline()'
                ]
            );
        }

        return parent::_prepareLayout();
    }
}

```

**Comment by engcom-Echo** [comment]
Hello @4DallasC 

Thank you for your report and collaboration!
We are not able to reproduce this issue on the 2.3.3 branch by provided steps.

**Testing scenario:**

**Preconditions:**
Simple and configurable products has been created and assigned to a category

**Scenario 1**
**Steps to reproduce:**
       

1. Add Simple Product to cart.
2. Place an order.
3. Go to admin.
4. Go to Sales → Orders and open the current order.
5. Go to Invoice.
6. Press "Submit Invoice".
7. Go to Credit Memo.
8. Press "Refund Offline".

**Result:** 
The Credit Memo has been created successfully. No errors. 
 ![Screenshot from 2019-11-22 11-29-22](https://user-images.githubusercontent.com/51681435/69428301-868e6f80-0d39-11ea-91c9-c7085a1b53aa.png)

![Screenshot from 2019-11-22 15-03-31](https://user-images.githubusercontent.com/51681435/69428253-69f23780-0d39-11ea-81b2-763c6863852c.png)

**Scenario 2**
**Steps to reproduce:**
       
1. Add Configurable Product to cart.
2. Place an order.
3. Go to admin.
4. Go to Sales → Orders and open the current order.
5. Go to Invoice.
6. Press "Submit Invoice".
7. Go to Credit Memo.
8. Press "Refund Offline".

**Result:** 
The Credit Memo has been created successfully. No errors.
![Screenshot from 2019-11-22 14-51-36](https://user-images.githubusercontent.com/51681435/69427741-8641a480-0d38-11ea-986b-3c1bc2c56e90.png)
![Screenshot from 2019-11-22 14-58-25](https://user-images.githubusercontent.com/51681435/69427870-a8d3bd80-0d38-11ea-8133-2c5114c76057.png)
Thank you!

**Comment by 4DallasC** [comment]
Hello @engcom-Echo 

Is the system you setup globally accessible? As I said in the beginning, that the page requires prototype to be included injected into the page for this function to work and is not always being included, but is sometimes, for example when one of the product has options.

To use legacy functions like `disableElements(\'submit-button\');` should be called like `require(['prototype'],function(){disableElements(\'submit-button\');});` to make sure it is included.

**Comment by ananth-iyer** [comment]
While creating a New Credit Memo/Refund, in item grid section there is "Return to Stock" checkbox. If you select that then Quantity will be directly added to the product.
![2019_11_23_Selection_01](https://user-images.githubusercontent.com/24892273/69474058-9cf00600-0de1-11ea-81c6-d51667f5bd7e.png)

So I don't think this is an issue in Magento 2.

**Comment by mystix** [comment]
> Hello @4DallasC
> 
> Thank you for your report and collaboration!
> We are not able to reproduce this issue on the 2.3.3 branch by provided steps.
> 
> **Testing scenario:**
> 
> **Preconditions:**
> Simple and configurable products has been created and assigned to a category
> 
> **Scenario 1**
> **Steps to reproduce:**
> 
> 1. Add Simple Product to cart.
> 2. Place an order.
> 3. Go to admin.
> 4. Go to Sales → Orders and open the current order.
> 5. Go to Invoice.
> 6. Press "Submit Invoice".
> 7. Go to Credit Memo.
> 8. Press "Refund Offline".
> 
> **Result:**
> The Credit Memo has been created successfully. No errors.
> ![Screenshot from 2019-11-22 11-29-22](https://user-images.githubusercontent.com/51681435/69428301-868e6f80-0d39-11ea-91c9-c7085a1b53aa.png)
> 
> ![Screenshot from 2019-11-22 15-03-31](https://user-images.githubusercontent.com/51681435/69428253-69f23780-0d39-11ea-81b2-763c6863852c.png)
> 
> **Scenario 2**
> **Steps to reproduce:**
> 
> 1. Add Configurable Product to cart.
> 2. Place an order.
> 3. Go to admin.
> 4. Go to Sales → Orders and open the current order.
> 5. Go to Invoice.
> 6. Press "Submit Invoice".
> 7. Go to Credit Memo.
> 8. Press "Refund Offline".
> 
> **Result:**
> The Credit Memo has been created successfully. No errors.
> ![Screenshot from 2019-11-22 14-51-36](https://user-images.githubusercontent.com/51681435/69427741-8641a480-0d38-11ea-986b-3c1bc2c56e90.png)
> ![Screenshot from 2019-11-22 14-58-25](https://user-images.githubusercontent.com/51681435/69427870-a8d3bd80-0d38-11ea-8133-2c5114c76057.png)
> Thank you!

@ananth-iyer I think you missed the point completely.

I just encountered the same javascript issue as @4DallasC on Magento 2.3.2-p2.

Steps to reproduce by @engcom-Echo are **incorrect**.
Instead of creating a credit memo from the Invoice view,
1) view the order
2) click Credit Memo from this order view
3) from the "New Memo" view, click "Refund Offline"
4) the following javascript error will be thrown:
`(index):624 Uncaught ReferenceError: disableElements is not defined
    at HTMLButtonElement.onclick ((index):624)`

Please reopen this bug report because the problem is real on 2.3.2-p2.

In my case, I was forced to create a new credit memo from the order view because I had a zero subtotal order.

**Comment by 4DallasC** [comment]
Hello All,

The code in the onclick events are dependent on prototype being injected into the page from the an unrelated script in vendor/magento/module-sales/view/adminhtml/templates/order/creditmemo/create/totals/adjustments.phtml. 

It's a legacy function call that requires a deprecated library and is dependent on another unrelated script injecting that deprecated library into the page. 

I give up ever reporting issues if you don't think this is an error.

**Comment by engcom-Echo** [comment]
Hello All

The bug was already fixed on 2.3-develop.

**Comment by fredden** [comment]
@engcom-Echo please can you provide a reference to the pull request or commit(s) where this has been fixed? We're encountering it on Magento 2.3.6.
